python/code_challenges/hashtable_repeated_word.py

```diff
@@ -1,5 +1,15 @@
-from data_structures.hashtable import Hashtable
+import re
 
 
-def first_repeated_word():
-    pass
+def first_repeated_word(string):
+    text_no_symbols = re.sub(r'[^\w]', ' ', string)
+    text_one_space = re.sub('\\s+', ' ', text_no_symbols)
+    text_lower = text_one_space.lower()
+    text_list = text_lower.split(" ")
+    helper_list = []
+
+    for word in text_list:
+        if word in helper_list:
```

Comment:
The time complexity of line 12 is O(n) per [https://wiki.python.org/moin/TimeComplexity](https://wiki.python.org/moin/TimeComplexity). This is inside a for loop (line 11). The combined time complexity is potentially very large!

The workaround to speed things up is to use a set/hashmap/dictionary for `helper_list`; all of these are O(1) lookups when checking if a key exists in helper_list. O(1) inside an O(n) loop is substantially faster!